pybamm/models/submodels/thermal/current_collector/base_current_collector_thermal.py

```diff
@@ -0,0 +1,220 @@
+#
+# Base class for thermal effects which accounts for current collectors
+#
+import pybamm
+
+
+class BaseModel(pybamm.BaseSubModel):
+    """Base class for thermal effects which accounts for current collectors.
+    Note: this can only be used for lumped, 1+1D or 2+1D thermal models in which
+    the temperature is independent of the through-cell direction x.
+
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+
+
+    **Extends:** :class:`pybamm.BaseSubModel`
+    """
+
+    def __init__(self, param):
+        super().__init__(param)
+
+    def _get_standard_fundamental_variables(self, T_av):
+        param = self.param
+
+        T_cn = T_av
+        T_n = pybamm.PrimaryBroadcast(T_av, ["negative electrode"])
+        T_s = pybamm.PrimaryBroadcast(T_av, ["separator"])
+        T_p = pybamm.PrimaryBroadcast(T_av, ["positive electrode"])
+        T_cp = T_av
+
+        # Note: T is the temperature defined only in the negative electrode,
+        # separator and positive electrode. This is the quanitity used in other
+        # submodels (e.g. electrolyte)
+        T = pybamm.Concatenation(T_n, T_s, T_p)
+
+        T_volume_av = self._yz_average(T_av)
+
+        q = self._flux_law(T)
+
+        variables = {
+            "Negative current collector temperature": T_cn,
```

Comment:
There is quite a bit of repeated code here from `base_thermal`. Its fine for now, but I think should be able to inherit from `base_thermal`